lib/index.coffee

```diff
@@ -51,7 +53,38 @@ streamToString = (stream) ->
 		.on 'end', ->
 			resolve(Buffer.concat(chunks).toString())
 
+describeCounterMetricIfNotExists = (metrics, name, description, labelNames) ->
+	if metrics.exists(name) == 'false'
+		metrics.describe.counter(name, description,
+			{
+				labelNames: labelNames
+			}
+		)
+
+describeHistogramMetricIfNotExists = (metrics, name, description, labelNames, buckets) ->
+	if metrics.exists(name) == 'false'
+		metrics.describe.histogram(name, description,
+			{
+				labelNames: labelNames,
+				buckets: buckets
+			}
+		)
+
+setupMetrics = (metrics) ->
+	describeCounterMetricIfNotExists(metrics, 'gc_space_reclaimed_total', 'Disk space the GC must free', ['host'])
+	describeCounterMetricIfNotExists(metrics, 'gc_number_images_to_remove_total', 'Number of images to be removed at a given time', ['host'])
+	describeCounterMetricIfNotExists(metrics, 'gc_image_removal_errors_total',  'Number of image removal errors by type', ['host', 'statusCode'])
+	describeCounterMetricIfNotExists(metrics, 'gc_images_removed_total', 'Total number of images removed by all different methods', ['host', 'removal_type'])
+	buckets = metrics.client.exponentialBuckets(4, Math.SQRT2, 29).map(Math.round)
+	describeHistogramMetricIfNotExists(metrics, 'gc_duration_seconds', 'Milliseconds taken to free GC the reclaimed space', ['host'], buckets)
```

Comment:
```suggestion
	describeHistogramMetricIfNotExists(metrics, 'gc_duration_milliseconds', 'Milliseconds taken to free GC the reclaimed space', ['host'], buckets)
```

I should have mentioned this, but i think we slightly prefer millisecond units for custom metrics (saving some operations to convert). 

ref https://github.com/balena-io/balena-api/blob/a71dbfd0b37509ef859bbfa2622c7f27b763a67b/src/metrics/describe.ts#L41-L58 & https://github.com/balena-io/balena-api/pull/1890#discussion_r336567786

cc @Page- & @dt-rush

Reply:
yep, 100

Reply:
Yeah, for me this is confusing. I was following the naming best practices as you suggested. I'll fix it :+1: